### PR TITLE
3.4 nan skip value

### PR DESCRIPTION
**Pull request by mneedham**

### CODE REVIEW

**Comment by mneedham** [comment]
@davidoliverSP2 I updated these to explain how the `skipValue` works and why the user would care about it.

File: doc/asciidoc/similarity-cosine.adoc [hunk 1]
```diff
@@ -185,10 +186,42 @@ include::scripts/similarity-cosine.cypher[tag=query]
...
 |===
 // end::query[]
 
+[[algorithms-similarity-cosine-cypher-projection]]
+== Cypher projection
+
+If the similarity lists are very large they can take up a lot of memory.
+For cases where those lists contain lots of values that should be skipped, you can use the less memory intensive approach of using Cypher statements to project the graph instead.
+
+.Set `graph:'cypher'` in the config:
```
Comment (davidoliverSP2) [inline doc/asciidoc/similarity-cosine.adoc]: the title of this example says `graph:'cypher'`, but the example uses `graph: "cypher"`

File: doc/asciidoc/similarity-cosine.adoc [hunk 2]
```diff
@@ -89,23 +89,23 @@ include::scripts/similarity-cosine.cypher[tag=stream]
 .Results
 [opts="header"]
 |===
-| `from`       | `to`         | `similarity`
```
Comment (davidoliverSP2) [inline doc/asciidoc/similarity-cosine.adoc]: it looks like this header row has been removed accidentally

File: doc/asciidoc/similarity-cosine.adoc [hunk 3]
```diff
@@ -144,17 +145,17 @@ include::scripts/similarity-cosine.cypher[tag=stream-topk]
...
+| "Michael"  | "Arya"     | 0.9788908326303921
+| "Praveena" | "Karin"    | 1.0
+| "Zhen"     | "Michael"  | 0.9542262139256075
 
 |===
 // end::stream-topk[]
 
 These results will not be symmetrical.
-For example, the person most similar to Praveena is Michael, but the person most similar to Zhen is also Michael.
+For example, the person most similar to Zhen is Michael, but the person most similar to Michael Arya.
```
Comment (davidoliverSP2) [inline doc/asciidoc/similarity-cosine.adoc]: missing word. 

Should this be "the person most similar to Michael **is** Arya"?

File: doc/asciidoc/similarity-euclidean.adoc [hunk 4]
```diff
@@ -84,24 +84,24 @@ include::scripts/similarity-euclidean.cypher[tag=stream]
...
+| "Michael"  | "Arya"     | 5.0
+| "Zhen"     | "Praveena" | 6.708203932499369
+| "Michael"  | "Karin"    | 7.0
+| "Praveena" | "Arya"     | 8.0
 
 |===
 // end::stream[]
 
-Arya and Karin have the most similar food preferences, with a euclidean distance of 7.68.
+Zhen and Arya and Zhen and Karin have the most similar food preferences, with a euclidean distance of 0.0.
```
Comment (davidoliverSP2) [inline doc/asciidoc/similarity-euclidean.adoc]: this sentence needs a comma:

Zhen and Arya, and Zhen and Karin have the most similar food preferences...

File: doc/asciidoc/similarity-euclidean.adoc [hunk 5]
```diff
@@ -140,18 +137,17 @@ include::scripts/similarity-euclidean.cypher[tag=stream-topk]
 .Results
 [opts="header",cols="1,1,1"]
 |===
-| `from`     | `to`       | `similarity`
```
Comment (davidoliverSP2) [inline doc/asciidoc/similarity-euclidean.adoc]: this header row looks like it's been removed by accident

File: doc/asciidoc/similarity-euclidean.adoc [hunk 6]
```diff
@@ -181,10 +177,42 @@ include::scripts/similarity-euclidean.cypher[tag=query]
...
 |===
 // end::query[]
 
+[[algorithms-similarity-euclidean-cypher-projection]]
+== Cypher projection
+
+If the similarity lists are very large they can take up a lot of memory.
+For cases where those lists contain lots of values that should be skipped, you can use the less memory intensive approach of using Cypher statements to project the graph instead.
+
+.Set `graph:'cypher'` in the config:
```
Comment (davidoliverSP2) [inline doc/asciidoc/similarity-euclidean.adoc]: see comment above about syntax

**Pull request merged**